java-squid/src/main/java/org/sonar/java/se/symbolicvalues/BinaryRelation.java

```diff
@@ -73,7 +113,7 @@ public boolean equals(Object obj) {
 
   private boolean equalsRelation(BinaryRelation rel) {
     if (kind.equals(rel.kind)) {
-      return key[0] == rel.key[0] && key[1] == rel.key[1];
+      return leftOp == rel.leftOp && rightOp == rel.rightOp;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)